test/unit/iam-assume-authenticator.test.js

```diff
@@ -0,0 +1,138 @@
+
+    // Should also create a token manager. Note that the options like `iamProfileId`
+    // aren't stored in the authenticator, but in the token manager.
+    expect(authenticator.tokenManager).toBeInstanceOf(IamAssumeTokenManager);
+    expect(authenticator.tokenManager.iamProfileId).toBe(config.iamProfileId);
+  });
+
+  it('should throw an error when no api key is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator();
+    }).toThrow('Missing required parameters: apikey');
+  });
+
+  it('should throw an error when no profile information is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({ apikey: config.apikey });
+    }).toThrow(
+      'Exactly one of `iamProfileName`, `iamProfileCrn`, or `iamProfileId` must be specified.'
+    );
+  });
+
+  it('should throw an error when too much profile information is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileId: 'some-id',
+        iamProfileCrn: 'some-crn',
+      });
+    }).toThrow(
+      'Exactly one of `iamProfileName`, `iamProfileCrn`, or `iamProfileId` must be specified.'
+    );
+  });
+
+  it('should throw an error when a profile name is provided without an account id', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileName: 'some-id',
+      });
+    }).toThrow('`iamProfileName` and `iamAccountId` must be provided together, or not at all');
+  });
+
+  it('should throw an error when an account id is provided without a profile name', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileId: 'some-id',
+        iamAccountId: 'some-account',
+      });
+    }).toThrow('`iamProfileName` and `iamAccountId` must be provided together, or not at all');
+  });
+
+  // "End to end" style test, to make sure this authenticator ingregates properly with parent classes.
+  it('should update the options and resolve with `null` when `authenticate` is called', async () => {
+    const authenticator = new IamAssumeAuthenticator({
+      apikey: config.apikey,
+      iamProfileId: config.iamProfileId,
+    });
+
+    // Override the created token manager with the mocked one.
+    authenticator.tokenManager = mockedTokenManager;
+
+    const options = { headers: { 'X-Some-Header': 'user-supplied header' } };
+    const result = await authenticator.authenticate(options);
+
+    expect(result).toBeUndefined();
+    expect(options.headers.Authorization).toBe(`Bearer ${fakeToken}`);
+    expect(getTokenSpy).toHaveBeenCalled();
+
+    // Verify that the original options are kept intact.
+    expect(options.headers['X-Some-Header']).toBe('user-supplied header');
+  });
+
+  it('should return the refresh token stored in the token manager', () => {
```

Comment:
We're not actually supporting a refresh token returned by the getToken "assume" operation.  It's part of the response, but the new authenticator doesn't (or probably shouldn't) expose the refresh token.

Reply:
As discussed on Slack, the refresh token will still be "available" but it will not be stored when we process the response. I've removed this test.